### ISSUE TITLE
Enhancement: [no-unsafe-call] Receive an error/warning when .tsconfig & .eslintrc ECMAScript versions are not matching. ex. .replaceAll()

### ISSUE THREAD

**Issue by BboyAkers** [typescript-eslint/typescript-eslint#5922]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/no-unsafe-call/

### Description

Many developers use typescript and eslint. This would definitely make for a better debugging experience if we receive errors stemming from the .tsconfig and the .eslintrc ECMAScript version not matching. When using .replaceAll() I received a "no-unsafe-call" error from eslint even I was using .replaceAll() on a variable that was a string.
### Fail

```typescript
.tsconfig
{
  "compilerOptions": {
    "target":"es2021"
}

.eslintrc
{
  "env": {
    "browser": true,
    "node": false
    //no "es2021":true added here
  },
}


const colors = {$palette-blue-10:"#d7eaf8",$palette-blue-20:"#9ccbee",$palette-blue-30:"#75b6e7"};

const formatColorPalette = (palette: Record<string, string>) => {
	return Object.keys(palette).reduce((acc, key) => {
		const oldKey = key;
		key = key.replace('$palette-', '');
                // error message that should be shown when hovering over the .replaceAll()
                // tsconfig 'target' and eslint env do not match. Update target and env to same ECMAScript version ex. es2021
		key = key.replaceAll('-', '');
		acc[key] = palette[oldKey];
		return acc;
	}, {} as Record<string, string>);
};

console.log(formatColorPalette(colors));
```

### Pass

```typescript
.tsconfig
{
  "compilerOptions": {
    "target":"es2021"
}

.eslintrc
{
  "env": {
    "browser": true,
    "node": false,
    "es2021": true
  },
}


const colors = {$palette-blue-10:"#d7eaf8",$palette-blue-20:"#9ccbee",$palette-blue-30:"#75b6e7"};

const formatColorPalette = (palette: Record<string, string>) => {
	return Object.keys(palette).reduce((acc, key) => {
		const oldKey = key;
		key = key.replace('$palette-', '');
                // error message that should be shown when hovering over the .replaceAll()
                // tsconfig 'target' and eslint env do not match. Update target and env to same ECMAScript version ex. es2021
		key = key.replaceAll('-', '');
		acc[key] = palette[oldKey];
		return acc;
	}, {} as Record<string, string>);
};

console.log(formatColorPalette(colors));
```

### Additional Info

- [X] I am willing to submit a pull request to implement this rule.

I'm not sure where to start but I'd love to get some pointers, dive in, and figure out how to make this PR happen! 🙂

**Comment by bradzacher** [comment]
We don't use the ecmascript version for anything important in this project and in fact we're about to remove those last usages in the next major.
Type-aware rules rely solely on what's in your tsconfig.

I'm not sure what your request is here - could you please elaborate?

**Comment by BboyAkers** [comment]
@bradzacher my apologies for my late response! I received an 'any' error message when using .replaceAll(). There was no specific error telling me that I needed to update my .eslintrc to match my tsconfig compiler options target. I'd like to know if it's possible to create a detailed error message stating this? If it is possible(and seems necessary) I'd be more than happy to try to dig into this and create a PR 🙂.

**Comment by bradzacher** [comment]
@BboyAkers 
As I mentioned - the lint rule you mentioned (`no-unsafe-call`) does not do anything with the `parserOptions.ecmaVersion`, nor does it do anything with the ESLint environment config.
Changing the options in your eslint config should have no impact on the rule whatsoever.

Could you please provide a reproduction for the problem you're running into using [our playground](https://typescript-eslint.io/play/)?

**Comment by Josh-Cena** [comment]
Closing because it sounds like a question
